core/src/main/java/hudson/slaves/JNLPLauncher.java

```diff
@@ -149,7 +149,9 @@ public boolean isLaunchSupported() {
         return false;
     }
 
-    @Deprecated
+    /**
+     * Deprecated (only used with deprecated {@code -jnlpUrl} mode), but cannot mark it as such without breaking CasC.
```

Comment:
This has the unfortunate effect that it is hard to notice improper usages of this method from plugins (for example does not show up as an IDE warning). I wonder if it would be better to just clean up and let anyone with an outdated JCasC file decide whether to clean it up too or tolerate deprecations: https://github.com/jenkinsci/jenkins/pull/9903#issuecomment-2640057543